### PR TITLE
feat: add jenkins job select in create-blueprint page

### LINKED ISSUE #3296: [Feature][jenkins] Jenkins uses `job` as the `scope`
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

For a long long time, we accepted the fact that Jenkins doesn't have scopes, this creates many if not countless problems in terms of System and UI design...

In a recent discussion, we realized that the `job` inside `Jenkins` could be perceived as data scopes.

### Use case

1. This is required for project related metric like dora
2. config-ui could be more unified

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by likyh**
# Summary

image
image

### Does this close any open issues?
Closes #3296

### CODE REVIEW

**Comment by e2corporation** [comment]
@likyh Since we have Jenkins using a "Job" concept to represent project, lets create `models/JenkinsJob.js` (modeled after `GithubProject` for example)

Something along the lines of...
```
class JenkinsJob {
  constructor(data = {}) {
    this.id = data?.id || null
    this.key = data?.key || this.id || null
    this.name = data?.name || null
    this.value = data?.value || this.name || this.id || null
    this.title = data?.title || this.name || this.id || null
    this.shortTitle = data?.shortTitle || null

    this.useApi = data?.useApi || true
    this.variant = data?.variant || 'job'
  }

  get(property) {
    return this[property]
  }

  set(property, value) {
    this[property] = value
    return this.property
  }

  getConfiguredEntityId() {
    return this.name.toString() || this.id
  }
}

export default JenkinsJob
```

After creating the model, search for `new GitHubProject` in the codebase and follow the same pattern to wrap Job objects with the `new JenkinsJob` model. The model could even be applied in one place at the `createListData` entry point, but you can follow the existing pattern being used by the other models for now.

File: config-ui/src/hooks/useDataScopesManager.jsx [hunk 1]
```diff
@@ -365,38 +365,64 @@ function useDataScopesManager({
...
+          })
+      ),
+    []
+  )
+
+  const getJenkinsProjects = useCallback(
+    (c) =>
+      c.scope.map(
+        (s) =>
+          new GitlabProject({
```
Comment (e2corporation) [inline config-ui/src/hooks/useDataScopesManager.jsx:400]: See comment about creating new `JenkinsJob` Model and use it here instead.

File: config-ui/src/components/blueprints/JenkinsJobsSelector.jsx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+            })
+          }}
+          onItemSelect={(item) => {
+            onItemSelect((rT) => {
+              return !rT[configuredConnection.id].includes(item)
+                ? {
+                    ...rT,
+                    [configuredConnection.id]: [
+                      ...rT[configuredConnection.id],
+                      item
```
Comment (e2corporation) [inline config-ui/src/components/blueprints/JenkinsJobsSelector.jsx:144]:
```suggestion
                      new JenkinsJob(item)
```

File: config-ui/src/hooks/useDataScopesManager.jsx (hunk 1 above)
Comment (likyh) [reply config-ui/src/hooks/useDataScopesManager.jsx:400]: I'll use JenkinsJob now. And I'll fix them to extend the base model when merge project and board.

File: config-ui/src/hooks/useDataScopesManager.jsx [hunk 3]
```diff
@@ -15,19 +15,17 @@
...
+import {integrationsData} from '@/data/integrations'
 import TransformationSettings from '@/models/TransformationSettings'
 import JiraBoard from '@/models/JiraBoard'
 import GitHubProject from '@/models/GithubProject'
 import GitlabProject from '@/models/GitlabProject'
-import { Providers, ProviderLabels, ProviderIcons } from '@/data/Providers'
-import { DataScopeModes } from '@/data/DataScopes'
+import {ProviderIcons, ProviderLabels, Providers} from '@/data/Providers'
+import {DataScopeModes} from '@/data/DataScopes'
+import JenkinsJob from "@/models/JenkinsJob";
```
Comment (e2corporation) [inline config-ui/src/hooks/useDataScopesManager.jsx:28]: A lint run needs to be run locally, not sure why Github CI didn't detect this line. Semi-colons should be stripped and double quote `->` single quote.

File: config-ui/src/components/blueprints/JenkinsJobsSelector.jsx [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+import React, { useEffect, useState } from 'react'
+import {
+  Checkbox,
+  Intent,
+  MenuItem,
+  Position,
+  Tooltip
+} from '@blueprintjs/core'
+import { MultiSelect } from '@blueprintjs/select'
+import GitlabProject from '@/models/GitlabProject'
```
[the remark this answers is not in the document]
Comment (likyh) [reply config-ui/src/components/blueprints/JenkinsJobsSelector.jsx]: fixed

**Comment by e2corporation** [comment]
@likyh Now that Jenkins won't depend on a one-off custom `C#X` Configuration key, have you tested Deployment Options workflow for Jenkins to ensure that transformation options are set correctly? (Both Create and Edit Modes)

**Comment by likyh** [comment]
> @likyh Now that Jenkins won't depend on a one-off custom `C#X` Configuration key, have you tested Deployment Options workflow for Jenkins to ensure that transformation options are set correctly? (Both Create and Edit Modes)

Yes. I run it after create and edit. It works well. 
image

File: config-ui/src/models/JenkinsJob.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+/**
+ * @typedef {object} GitlabProject
```
[the remark this answers is not in the document]
Comment (likyh) [reply config-ui/src/models/JenkinsJob.js]: fixed

**Review by e2corporation — APPROVED** [review]
@likyh Great work, LGTM `@config-ui` (after rebase)

**Review by mappjzc — APPROVED** [review]

**Pull request merged**